src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -156,11 +171,12 @@ private async Task SynchronizeSuppressedIssuesAsync()
                 var moduleKeyToRelativePathToRoot = (await this.sonarQubeService.GetAllModulesAsync(sonarQubeProjectKey,
                         cancellationTokenSource.Token))
                     .ToDictionary(x => x.Key, x => x.RelativePathToRoot ?? string.Empty);
+                this.hasModules = moduleKeyToRelativePathToRoot.Keys.Count == 1;
 
                 this.cachedSuppressedIssues = (await this.sonarQubeService.GetSuppressedIssuesAsync(sonarQubeProjectKey,
                         cancellationTokenSource.Token))
-                    .Select(x => ToIssueWithFixedRelativePath(moduleKeyToRelativePathToRoot, x))
-                    .GroupBy(x => x.FilePath)
+                    .Select(x => new { Key = ProcessKey(moduleKeyToRelativePathToRoot, x), Issue = x })
+                    .GroupBy(x => x.Key, x => x.Issue)
                     .OrderBy(x => x.Key.Length)
```

Comment:
Don't we want to sort by key length in descending order? (default sort order is ascending)

If so, then there's a test hole: we need to test having the same file in different directories.